Foundation/NSStream.swift

```diff
@@ -209,20 +227,32 @@ public class NSOutputStream : Stream {
         return self.init(toMemory: ())
     }
     
-    public override func propertyForKey(_ key: String) -> AnyObject? {
-        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    public override var streamError: NSError?{
+        let error = CFWriteStreamCopyError(_stream)
+        return error?._nsObject
     }
     
     public  override func setProperty(_ property: AnyObject?, forKey key: String) -> Bool {
+        guard let property = property else { return false }
         return CFWriteStreamSetProperty(_stream, key._cfObject, property)
     }
+    
+    public override func propertyForKey(_ key: String) -> AnyObject? {
+        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    }
+    
 }
 
 // Discussion of this API is ongoing for its usage of AutoreleasingUnsafeMutablePointer
 #if false
 extension Stream {
+    //Currently not tested as it is compiled out through the macro setting false
```

Comment:
the function signature could experimentally be re-written as:

```swift
public class func getStreamsToHost(withName hostName: String, port: Int) -> (inputStream: InputStream?, outputStream: OutputStream?)
// or
public class func getStreamsToHost(withName hostName: String, port: Int, inputStream: inout InputStream?, outputStream: inout OutputStream?)
```

However this will cause an allocation of a stream that is not needed in both cases. The second would look more similar to call sites